### PR TITLE
Update docs for `secretsMap` option support

### PR DESCRIPTION
**Pull request by Huanli-Meng**
### Motivation
In Function Mesh V0.1.9 release, [secrets are supported](https://github.com/streamnative/function-mesh-worker-service/pull/21). Therefore, update docs for this code PR.

### Modifications:
- Update docs for secrets supported on function mesh worker service.
  - Related release: latest
- Refine authentication descriptions 
  - Related releases: v0.1.5- latest
- Update typos, code format and heading levels:
  - Related releases: all releases

### CODE REVIEW

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (freeznet) [inline docs/connectors/io-crd-config/sink-crd-config.md:79]: `SecretsMap` is defined as a `Map` struct with `String` keys and `SecretReference` values. The `key` will indicate the environment value in the container, and the `SecretReference` is defined as below.

| Field | Description |
 | --- | --- |
 | `path` | The name of the secret in the pod's namespace to select from. |
 | `key` | The key of the secret to select from.  Must be a valid secret key. |

For example, the user has a Kubernetes Secret named `credential-secret` defined as below:

```yaml
 apiVersion: v1
 data:
   username: foo
   password: bar
 kind: Secret
 metadata:
   name: credential-secret
 type: Opaque
 ```

To use it in Pulsar Functions in a security way, simply define the `SecretsMap` in the Custom Resource:

```
secretsMap:
  username:
    path: credential-secret
    key: username
  password:
    path: credential-secret
    key: password
```

Then in the Pulsar Functions and Connectors, you can call `context.getSecret("username")` to get the secret value.

**Review by freeznet — COMMENTED** [review]
please check my comment, I only made suggestions on sink doc and the changes can be used for source and function as well because the content of sink, source and function should be similar.

**Comment by Huanli-Meng** [comment]
@freeznet Thanks for your valuable comments. Docs have been updated based on you comments. PTAL again and feel free to approve it if there is no more comment. Thanks.

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (freeznet) [inline docs/connectors/io-crd-config/sink-crd-config.md:94]:
```suggestion
    key: username
```

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (freeznet) [inline docs/connectors/io-crd-config/sink-crd-config.md:97]:
```suggestion
    key: password
```

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (freeznet) [inline docs/connectors/io-crd-config/sink-crd-config.md:100]:
```suggestion
Then, in the Pulsar Functions and Connectors, you can call `context.getSecret("username")` to get the secret value `"foo"`.
```

**Review by freeznet — COMMENTED** [review]
sorry, i have made some minor changes, to fix the wrong resouce ref in the custom resource.

**Comment by Huanli-Meng** [comment]
> sorry, i have made some minor changes, to fix the wrong resouce ref in the custom resource.

It's OK. I have updated the doc based on your comments. PTAL again. Thanks

**Review by freeznet — APPROVED** [review]
thanks, lgtm

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline docs/connectors/io-crd-config/sink-crd-config.md:66]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..") and a comma before which in the last sentence "... `EnvironmentBasedSecretsProvider`, which..."

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline docs/connectors/io-crd-config/sink-crd-config.md:88]: "in a secure way"

File: docs/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline docs/connectors/io-crd-config/sink-crd-config.md:104]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: docs/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline docs/connectors/io-crd-config/source-crd-config.md:59]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..") and a comma before which in the last sentence "... EnvironmentBasedSecretsProvider, which..."

File: docs/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline docs/connectors/io-crd-config/source-crd-config.md:97]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: docs/functions/function-crd.md
Comment (teachwrite) [inline docs/functions/function-crd.md:87]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..") and a comma before which in the last sentence "... EnvironmentBasedSecretsProvider, which..."

File: docs/functions/function-crd.md
Comment (teachwrite) [inline docs/functions/function-crd.md:109]: "in a secure way"

File: versioned_docs/version-0.1.5/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.5/connectors/io-crd-config/sink-crd-config.md:65]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.5/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.5/connectors/io-crd-config/source-crd-config.md:57]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.5/functions/function-crd.md
Comment (teachwrite) [inline versioned_docs/version-0.1.5/functions/function-crd.md:86]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.6/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.6/connectors/io-crd-config/sink-crd-config.md:65]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.6/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.6/connectors/io-crd-config/source-crd-config.md:57]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.6/functions/function-crd.md
Comment (teachwrite) [inline versioned_docs/version-0.1.6/functions/function-crd.md:86]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.7/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.7/connectors/io-crd-config/sink-crd-config.md:66]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.7/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.7/connectors/io-crd-config/source-crd-config.md:59]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.7/functions/function-crd.md
Comment (teachwrite) [inline versioned_docs/version-0.1.7/functions/function-crd.md:87]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.7/scaling.md
Comment (teachwrite) [inline versioned_docs/version-0.1.7/scaling.md:47]: "If these metrics do not meet..."

File: versioned_docs/version-0.1.8/connectors/io-crd-config/sink-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.8/connectors/io-crd-config/sink-crd-config.md:66]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.8/connectors/io-crd-config/source-crd-config.md
Comment (teachwrite) [inline versioned_docs/version-0.1.8/connectors/io-crd-config/source-crd-config.md:59]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.8/functions/function-crd.md
Comment (teachwrite) [inline versioned_docs/version-0.1.8/functions/function-crd.md:87]: need a comma after "Source" in the first sentence ("...Function, Source, and Sink..")

File: versioned_docs/version-0.1.8/scaling.md
Comment (teachwrite) [inline versioned_docs/version-0.1.8/scaling.md:47]: "If these metrics do not meet..."

**Review by teachwrite — CHANGES_REQUESTED** [review]
These are all small changes, and many are changes to the same sentence repeated through the documents. After you make these changes, you can go ahead and merge this PR.

**Comment by Huanli-Meng** [comment]
@teachwrite , I have applied your comments into docs. PTAL again. If you choose "Change request" when you submit your comments, i cannot merge the PR directly. Thanks.

**Pull request merged**
